src/components/GlobalFilter/tagsApi.ts

```diff
@@ -70,38 +73,109 @@ export function getAllTags({ search, activeTags, registeredWith }: TagFilterOpti
     // @ts-ignore
     registeredWith ? registeredWith : undefined,
     undefined,
-    {
-      query: generateFilter(buildFilter(workloads, SID)),
-    }
+    { query: generateFilter(buildFilter(workloads, SID)) }
   );
+
+  // @ts-ignore
+  chromeStore.set(tagsAtom, (prev) => ({
+    ...prev,
+    isLoaded: true,
+    // Update pagination state
+    page: pagination?.page || 1,
+    perPage: pagination?.perPage || 10,
+    // Transform API results into the required GlobalFilterTag[] structure for TagsModal
+    items: [
+      {
+        id: 'tags-group',
+        name: 'Tags',
+        tags: ((response as any).data?.results || (response as any).results || [])
```

Comment:
Please do not use `as any`. You should typeguard the result variable somwhere above instead. And you wont have to constantly cast tit as any